### DISCUSSION TITLE
Different extra for different engines

### DISCUSSION THREAD

**Discussion by ghost** [SeaQL/sea-query#571] [Q&A]
I want to do something like `.extra(sqlite, "anything on sqlite").extra(postgres, "anything on postgres").extra(mysql, "anything on mysql")`. So that I don't have to choose an engine when I use `.extra()`.

**Comment by ikrivosheev** [comment]
@JediMaster25 hello! Yes, you cannot set extra for a specific backend.

**Comment by billy1624** [comment]
Hey @JediMaster25, I think you can set the `extra` for different engine. But it has to be done on your application code.

```rs
match db_engine {
    // append DB specific `extra`...
}
```

**Comment by ghost** [comment]
@billy1624 could you complete that example with a string for each of the three supported engines?

**Reply by billy1624** [reply]
Your application code has to be DB specific. I assumed you have a Enum to signify the target DB engine that the create table statement to be executed on.

```rs
enum DatabaseEngine {
    // ...
}

match db_engine {
    // append DB specific `extra`...
}
```
